src/components/Table/PTable.vue

```diff
@@ -138,19 +134,19 @@
     }
   }
 
-  function convertStringsToTableColumns(columns: string[]): TableColumn[] {
-    return columns.map(property => ({ label: property, property }))
+  function convertStringsToTableColumns(columns: string[]): TColumn[] {
+    return columns.map(property => ({ label: property, property })) as TColumn[]
   }
 
-  function getValue(row: TableData, column: TableColumn): unknown {
-    if (column.property) {
-      return row[column.property]
+  function getValue<P extends keyof TData>(row: TData, property: P | undefined): any {
```

Comment:
Warning here - is `any` correct? I'd expect to return a generic here

Reply:
ah I should have explained the reason for the `any`. I added an eslint-ignore and a comment for this. But basically this is for backwards compatibility. A column slot is based on the column label. Which does not have the restriction of being a `keyof TData`. It can be any string. If you add an optional `property: keyof TData` then it will auto populate the value and bind it to the slot scope. I feel like it might be possible to infer the type based on the `property` if there is one and hard code it to `undefined` if there is not. But that would take some typescript magic that I'm not prepared to wrangle right now lol.